core/io/json.h

```diff
@@ -105,6 +105,9 @@ class ResourceFormatLoaderJSON : public ResourceFormatLoader {
 	virtual void get_recognized_extensions(List<String> *p_extensions) const override;
 	virtual bool handles_type(const String &p_type) const override;
 	virtual String get_resource_type(const String &p_path) const override;
+	// Treat JSON as a text file, do not generate a `*.json.uid` file.
+	virtual ResourceUID::ID get_resource_uid(const String &p_path) const override { return ResourceUID::INVALID_ID; }
+	virtual bool has_custom_uid_support() const override { return true; }
```

Comment:
Note that **both** overrides are required. This may seem odd, since in fact JSON does not have a UID. But we define "support" for custom UID... which is invalid. It might make sense to separate these three lines with a blank line for better readability.

https://github.com/godotengine/godot/blob/f952bfe9985ad8f507cc29b2c7601bbba18b8039/core/io/resource_loader.cpp#L115-L126

https://github.com/godotengine/godot/blob/f952bfe9985ad8f507cc29b2c7601bbba18b8039/editor/editor_node.cpp#L1457-L1465

https://github.com/godotengine/godot/blob/f952bfe9985ad8f507cc29b2c7601bbba18b8039/editor/editor_file_system.cpp#L1263-L1270

Reply:
We could make it into a macro called `DISABLE_UID` or something. These 2 lines will be the same for every class.

Reply:
I think this is redundant for now. But we may reconsider this in the future if more exceptions are added.